manager/controllers/app/m4dapplication_controller.go

```diff
@@ -286,42 +282,45 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 	return ctrl.Result{}, nil
 }
 
-// ConstructIngestDataInfo already has the information about the external data set's endpoint,
-// but it adds the governance decisions. It also already has the link to the credentials, which was provided by the user.
+// ConstructNewDataInfo handles the flow where a new data set is being added to the M4D managed environment.  This means that the data set
+// needs to be cataloged in the environment's catalog.
 // NOTE: Assumes the credentials were put into the M4D credential manager by the user!!
-func (r *M4DApplicationReconciler) ConstructIngestDataInfo(externalDataset app.ExternalDataContext, req *modules.DataInfo, input *app.M4DApplication, clusters []multicluster.Cluster) error {
-	dataDetails := pb.DatasetDetails{
-		DataOwner:  "",
-		DataStore:  externalDataset.ExternalStore.Connection,
-		DataFormat: externalDataset.ExternalStore.Format,
-		Geo:        string(externalDataset.Residency),
-	}
-	// Store link to credentials rather than the credentials themselves
-	// This is a hack, but works for now.
+// TODO: When implementing multiple data catalogs there needs to be a way to differentiate between the "inline" catalog where the source data
+// info resides, and the catalog to which we want to register the data set.
+// TODO: Resolve credential issues
+func (r *M4DApplicationReconciler) ConstructNewDataInfo(dataset app.DataContext, req *modules.DataInfo, input *app.M4DApplication, clusters []multicluster.Cluster) error {
+	// Call the DataCatalog service to get info about the dataset
+	if err := GetConnectionDetails(dataset.DataSetID, dataset.DataCatalogID, req, input); err != nil {
+		return AnalyzeError(input, r.Log, dataset.DataSetID, err, "Catalog Connector")
+	}
+
+	// In cataloged data sets we are reading the actual credentials from the external credential manager.
+	// In this case we are assuming that the user put the credentials in our internal credential manager, and thus we only need to get the link.
+	// TODO: Need a way to get the link.  From where will we get it?  The internal data catalog?  Need a relevant API.  Temp code is a temporary place holder.
```

Comment:
I suggest the following scenario:
- Admin creates a secret (or several secrets) in m4d-system of the coordinator cluster. This secret holds the credentials. 
- Credential manager can access the secret (based on datasetId/catalogId)
- Credential manager returns the credentials to manager 
- Manager stores the credentials in vault and writes the path in a relevant blueprint.

Reply:
The data owner provides the credentials, not an admin.  If I understood @roee88 correctly, the plan is for the data owner to put the credentials directly into the M4D vault (like the data user does for the credentials to external systems such as the enterprise catalog) and put the link to them into the inline data catalog.  The manager then needs to get that link, which is passed to the modules in the blueprint.
@roee88 Did I understand correctly?

Reply:
If you mean VaultConnector that retrieves the credentials that the user has put in the user vault, then the connector will return the credentials, not the link. It will access the user vault instance according to the link, retrieve the credentials and pass to manager that will register them in the internal vault and pass to the modules. Data owner does not have an access to m4d-system and m4d vault. That's why VaultConnector uses an additional set of variables for indicating url, token, root (USER_VAULT_TOKEN, USER_VAULT_PATH, etc.)